src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,256 @@
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+abstract class NovelFullGlobal(
+    override val lang: String,
+    val supportsSearch: Boolean = true,
+) : ParsedHttpSource() {
+
+    override val baseUrl = "https://mangafre.com" // https://allwebnovel.com
+
+    override val name = "NovelFull"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder().build()
+
+    private val preferences: SharedPreferences =
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=hot_novel&page_num=$page&language=$lang", headers)
+    }
+
+    override fun popularMangaSelector(): String {
+        return "#list-page > .col-truyen-main .row"
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.select(Evaluator.Tag("a")).attr("alt")
+            element.selectFirst(Evaluator.Tag("img"))?.also { thumbnail_url = it.imgAttr() }
+            setUrlWithoutDomain(element.selectFirst(Evaluator.Tag("a"))!!.absUrl("href"))
+        }
+    }
+
+    override fun popularMangaNextPageSelector(): String? {
+        return ".pagination > .next"
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        if (page == 1) {
+            val url = "$baseUrl/comic/index.html?language=$lang"
+            fetchGenres(client, url, headers, preferences)
+        }
+        return super.fetchPopularManga(page)
+    }
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=last_release&page_num=$page&language=$lang", headers)
+    }
+
+    override fun latestUpdatesSelector(): String = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            return GET("$baseUrl/comic/search.html?keyword=$query&page_num=$page", headers)
+        }
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    val filterId = filter.selected
+                    Log.i("NovelFullGlobal_Genre", "Filtering by genre: $filterId")
+                    return GET("$baseUrl/comic/bookclass.html?type=category_novel&id=$filterId&page_num=$page&language=$lang", headers)
```

Comment:
Make sure `filterId` does not contain special characters. Preferably use `HttpUrlBuilder`.